server/socket/index.ts

```diff
@@ -1,147 +1,126 @@
-      log.trace(`Received message ${message} from session ${sessionId}`);
-      try {
-        const payload: IWebsocketsPayload<any> = JSON.parse(message.toString());
-        if (payload.event === WebsocketMessageType.Heartbeat) {
-          ws.isAlive = true;
-
-          // reset liveness timers
-          if (ws.aliveTimer) clearTimeout(ws.aliveTimer);
-          ws.aliveTimer = global.setTimeout(() => { ws.isAlive = false; }, ALIVE_TIMEOUT);
-
-          if (ws.expirationTimer) clearTimeout(ws.expirationTimer);
-          ws.expirationTimer = global.setTimeout(() => {
-            // TODO: do i need to manually close the socket here?
-            log.trace(`Websocket ${sessionId} expired, emitting close`);
-            wss.emit(WebsocketEventType.Close);
-          }, EXPIRATION_TIME);
+  io.engine.on('connection_error', (err) => {
+    // log.error(err.req);      // the request object
+    // console.log(err.code);     // the error code, for example 1
+    // console.log(err.message);  // the error message, for example "Session ID unknown"
+    // console.log(err.context);  // some additional error context
+    log.error('A WebSocket connection error has occurred', err);
+  });
 
-          // get user if verified
-          if (payload.jwt) {
-            jwt.verify(payload.jwt, JWT_SECRET, async (err, decodedUser) => {
-              if (err) {
-                log.info(`received message with malformed JWT: ${payload.jwt}`);
-              } else {
-                ws.isAuthenticated = true;
-                ws.user = decodedUser;
-              }
-            });
-          }
-        } else {
-          log.error('received malformed message');
-        }
-      } catch (e) {
-        log.error('received malformed message');
-      }
-    });
+  // create the chain-events namespace
+  const ceNamespace = createCeNamespace(io);
 
-    ws.on(WebsocketEventType.Close, () => {
-      log.trace(`Received close event for websocket ${sessionId}`);
-      if (sessionMap[sessionId]) {
-        delete sessionMap[sessionId];
-      }
-      if (userId && userMap[userId]) {
-        delete userMap[userId];
-      }
-    });
+  // enables the admin analytics dashboard (creates /admin namespace)
+  instrument(io, {
+    auth: false,
   });
 
-  // TODO: maybe unify these, or else remove the event type from payload and add it manually here?
-  wss.on(WebsocketMessageType.InitializeScrollback, (payload: IWebsocketsPayload<any>, userIds: number[]) => {
-    if (logging) log.info(`Payloading ${JSON.stringify(payload)} to users ${JSON.stringify(userIds)}`);
-    for (const user of userIds) {
-      if (user && user in userMap && userMap[user].isAlive) {
-        userMap[user].send(JSON.stringify(payload));
-      }
-    }
+  const pool = new Pool({
+    connectionString: DATABASE_URI,
+    ssl: {
+      rejectUnauthorized: false,
+    },
   });
+  pool
+    .query(
+      `
+          CREATE TABLE IF NOT EXISTS socket_io_attachments
```

Comment:
How/where exactly does this table get used? I don't see it referenced elsewhere.

Reply:
Used behind the scenes by Socket.io for synchronizing between Socket.io servers. The error you were getting for the PostgresAdapter is related to this (needs this table). Socket.io was trying to use SSL to connect to the db so that it could create this table. When that failed, Socket.io let us know that the adapter would not be working so synchronization between multiple nodes would not work. I just pushed the same SSL fix you did on `dbNode.ts` so it won't happen here again and table should be created regardless of local Postgres SSL status.